mixnet/config.py

```diff
@@ -2,110 +2,77 @@
 
 import random
 from dataclasses import dataclass
-from typing import List, TypeAlias
+from typing import List
 
 from cryptography.hazmat.primitives.asymmetric.x25519 import (
     X25519PrivateKey,
     X25519PublicKey,
 )
-from pysphinx.node import Node
-
-from mixnet.bls import BlsPrivateKey, BlsPublicKey
-from mixnet.fisheryates import FisherYates
+from pysphinx.sphinx import Node as SphinxNode
 
 
 @dataclass
-class MixnetConfig:
-    topology_config: MixnetTopologyConfig
-    mixclient_config: MixClientConfig
-    mixnode_config: MixNodeConfig
-
+class GlobalConfig:
+    """
+    Global parameters used across all nodes in the network
+    """
 
-@dataclass
-class MixnetTopologyConfig:
-    mixnode_candidates: List[MixNodeInfo]
-    size: MixnetTopologySize
-    entropy: bytes
+    membership: MixMembership
+    transmission_rate_per_sec: int  # Global Transmission Rate
+    # TODO: use this to make the size of Sphinx packet constant
+    max_mix_path_length: int
 
 
 @dataclass
-class MixClientConfig:
-    emission_rate_per_min: int  # Poisson rate parameter: lambda
-    redundancy: int
-    topology: MixnetTopology
+class NodeConfig:
+    """
+    Node-specific parameters
+    """
+
+    private_key: X25519PrivateKey
+    mix_path_length: int
+    gossip: GossipConfig
 
 
 @dataclass
-class MixNodeConfig:
-    encryption_private_key: X25519PrivateKey
-    delay_rate_per_min: int  # Poisson rate parameter: mu
+class GossipConfig:
+    # The target number of peers a node should maintain in its p2p network
+    peering_degree: int
 
 
 @dataclass
-class MixnetTopology:
-    # In production, this can be a 1-D array, which is accessible by indexes.
-    # Here, we use a 2-D array for readability.
-    layers: List[List[MixNodeInfo]]
-
-    def __init__(
-        self,
-        config: MixnetTopologyConfig,
-    ) -> None:
-        """
-        Build a new topology deterministically using an entropy and a given set of candidates.
-        """
-        shuffled = FisherYates.shuffle(config.mixnode_candidates, config.entropy)
-        sampled = shuffled[: config.size.num_total_mixnodes()]
+class MixMembership:
+    """
+    A list of public information of nodes in the network.
+    We assume that this list is eventually known to all nodes in the network (e.g. via p2p advertising).
```

Comment:
Makes sense. I'll remove the `(e.g. via p2p advertising)` and `eventually` parts because this information is probably going to be shared as an on-chain data. Since it is not determined 100% clearly, I won't mention anything specifically. Instead, just
```
We assume that this list is known to all nodes in the network.
```

Reply:
Great! Thank you!